backend/satellite_tools/repo_plugins/yum_src.py

```diff
@@ -630,15 +631,20 @@ def setup_repo(self, repo, uln_repo=False):
         zypper_cmd = "zypper"
         if not self.interactive:
             zypper_cmd = "{} -n".format(zypper_cmd)
-        ret_error = os.system("{} --root {} --reposd-dir {} --cache-dir {} --raw-cache-dir {} --solv-cache-dir {} ref".format(
+        zypper_cmd = "{} --root {} --reposd-dir {} --cache-dir {} --raw-cache-dir {} --solv-cache-dir {} ref".format(
             zypper_cmd,
             REPOSYNC_ZYPPER_ROOT,
             os.path.join(repo.root, "etc/zypp/repos.d/"),
             REPOSYNC_ZYPPER_RPMDB_PATH,
             os.path.join(repo.root, "var/cache/zypp/raw/"),
             os.path.join(repo.root, "var/cache/zypp/solv/")
-        ))
+        )
+        process = subprocess.run(zypper_cmd.split(' '), stderr=subprocess.PIPE)
+        ret_error = process.returncode
+
         if ret_error:
+            if errs:
```

Comment:
AFAICS this `errs` variable here is not defined at this point. Should this be `process.stderr` ?

Also, maybe could we do something like:

```python
    if process.returncode:
        if process.stderr:
            raise RepoMDError("Cannot access repository.\n{}".format(sstr(process.stderr)))
        raise RepoMDError("Cannot access repository. Maybe repository GPG keys are not imported")
```

to avoid declaring `ret_error = process.returncode` which seems no to be really necessary.

Reply:
changed